doc/symfony-cache-configuration.rst

```diff
@@ -179,4 +179,34 @@ the ``session_name_prefix`` option) in the requests to the backend. If you need
 a different behavior, overwrite ``UserContextSubscriber::cleanupHashLookupRequest``
 with your own logic.
 
+.. _symfony-cache x-debugging:
+
+Debugging
+~~~~~~~~~
+
+For the ``assertHit`` and ``assertMiss`` assertions to work, you need to add
+debug information in your AppCache. Create the cache kernel with the option
+``'debug' => true`` and add the following to your ``AppCache``::
+
+    public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQUEST, $catch = true)
+    {
+        $response = parent::handle($request, $type, $catch);
+
+        if ($response->headers->has('X-Symfony-Cache')) {
+            if (false !== strpos($response->headers->get('X-Symfony-Cache'), 'miss')) {
+                $state = 'MISS';
+            } elseif (false !== strpos($response->headers->get('X-Symfony-Cache'), 'fresh')) {
+                $state = 'HIT';
+            } else {
+                $state = 'UNDETERMINED';
+            }
+            $response->headers->set('X-Cache', $state);
+        }
+
+        return $response;
+    }
+
+The ``UNDETERMINED`` state should never happen. If it does, it means that your
+HttpCache is not correctly set into debug mode.
```

Comment:
HttpCache? also, should we make this into another subscriber that one can simply activate in app_test.php?